engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationProcessor.java

```diff
@@ -533,10 +533,15 @@ private void terminateElement(
     // terminate all child instances if the element is an event subprocess
     final BpmnElementType elementType = elementInstance.getValue().getBpmnElementType();
     if (elementType == BpmnElementType.EVENT_SUB_PROCESS
-        || elementType == BpmnElementType.SUB_PROCESS) {
+        || elementType == BpmnElementType.SUB_PROCESS
+        || elementType == BpmnElementType.PROCESS) {
       elementInstanceState.getChildren(elementInstanceKey).stream()
           .filter(ElementInstance::canTerminate)
           .forEach(childInstance -> terminateElement(childInstance, sideEffects));
+    } else if (elementType == BpmnElementType.CALL_ACTIVITY) {
+      final var calledActivityElementInstanceState =
+          elementInstanceState.getInstance(elementInstance.getCalledChildInstanceKey());
```

Comment:
🔧 I think the variable name accidentally got postfixed with `State`, but it should probably be named `childProcessInstance` anyways